### ISSUE TITLE
Can't install Arch on this new archinstall build with manual partitioning

### ISSUE THREAD

**Issue by Faugus** [archlinux/archinstall#1971]
![IMG_20230801_141145_1](https://github.com/archlinux/archinstall/assets/112667550/71e10fd7-c2ff-4b1c-8b16-e7661c07c836)

As I always did:
1gb - wipe - fat32 - /boot
100gb - wipe - ext4 - /

Edit: looks like it's not being able to change the filesystem of the partition. I've created 2 ext4 partitions by mistake with gnome-disk-utility and when I tried to install with archinstall, the boot partition stood as ext4 after failing.

**Comment by Torxed** [comment]
Could you post your user config?

**Comment by Faugus** [comment]
> Could you post your user config?

How can I save the config file if the system is not being installed?

**Comment by svartkanin** [comment]
@Faugus could you upload the log file located under /var/log/archinstall/install.log

**Comment by amebtw** [comment]
@Faugus Connect external hdd/stick that is formatted(ext4 for simplicity), mount it and then copy the contents of /var/log/archinstall to it.

**Comment by Faugus** [comment]
@Torxed @amebtw @svartkanin 
[archinstall-logs.tar.gz](https://github.com/archlinux/archinstall/files/12241605/archinstall-logs.tar.gz)

**Comment by Torxed** [comment]
Weird, `user_configuration.json` and `user_credentials.json` are binary blobs oÔ

`install.log` is intact however and contains the config too, so we're fine.

This appears broken:
```
Cannot import <module 'sway' from '/usr/lib/python3.11/site-packages/archinstall/default_profiles/desktops/sway.py'>, it does not appear to be a Profile class
```

This is the relevant disk configuration it appears:
```json
{
    "btrfs": [],
    "flags": [
        "Boot",
        "ESP"
    ],
    "fs_type": "fat32",
    "length": {
        "sector_size": null,
        "total_size": null,
        "unit": "B",
        "value": 1000341504
    },
    "mount_options": [],
    "mountpoint": "/boot",
    "obj_id": "d6d52877-af51-4729-8233-41a2b804004f",
    "start": {
        "sector_size": {
            "sector_size": null,
            "total_size": null,
            "unit": "B",
            "value": 512
        },
        "total_size": null,
        "unit": "sectors",
        "value": 236333056
    },
    "status": "modify",
    "type": "primary"
},
{
    "btrfs": [],
    "flags": [],
    "fs_type": "ext4",
    "length": {
        "sector_size": null,
        "total_size": null,
        "unit": "B",
        "value": 100000595968
    },
    "mount_options": [],
    "mountpoint": "/",
    "obj_id": "887f2e17-4f82-46c8-8c86-13994550e9d7",
    "start": {
        "sector_size": {
            "sector_size": null,
            "total_size": null,
            "unit": "B",
            "value": 512
        },
        "total_size": null,
        "unit": "sectors",
        "value": 238286848
    },
    "status": "modify",
    "type": "primary"
},
```

I'm taking a look at it during the day!

**Comment by Torxed** [comment]
I know it's asking a lot, but could OP or anyone else having this issue record their steps?
Either in a VM with:
```
$ giph -s -f 20 ~/archinstall.mkv
```
Or just a cellphone would be nice.
Mainly from the main menu all the way through partitioning and pressing "Install".

The reason being that I can't reproduce this, and going through what could go wrong in the code might take a while.

**Comment by Faugus** [comment]
@Torxed https://youtu.be/CoZRThU5wGA :)

**Comment by amebtw** [comment]
The problem is not reproducible in all runs. I made a first run and it gave the error. I ran it a second time, without rebooting, just archinstall, and it went to finish, but no system was found after reboot.

**Comment by Torxed** [comment]
Awesome work, thank you for that solid recording!
I noticed something, the first partition has no fstype:
```json
{
    "btrfs": [],
    "flags": [],
    "fs_type": "",
    "length": {
        "sector_size": null,
        "total_size": null,
        "unit": "B",
        "value": 20000538624
    },
    "mount_options": [],
    "mountpoint": null,
    "obj_id": "8f4c7037-dd84-438c-9dfc-f7e7b8818a06",
    "start": {
        "sector_size": {
            "sector_size": null,
            "total_size": null,
            "unit": "B",
            "value": 512
        },
        "total_size": null,
        "unit": "sectors",
        "value": 2048
    },
    "status": "existing",
    "type": "primary"
},
```

I wonder if that's what it's crashing on? Although it shouldn't use it, perhaps it's trying to validate it despite not having any modification requests...

I'll look in to that.

**Comment by codefiles** [comment]
The `format()` function passes a `DeviceModification` to `_validate()`.

https://github.com/archlinux/archinstall/blob/81ef09c336c1fc4dfbe5a324349d95a6bb30ce6d/archinstall/lib/disk/device_handler.py#L260-L269

This validates the `DeviceModification.partitions`, a `List[PartitionModification]`.

https://github.com/archlinux/archinstall/blob/81ef09c336c1fc4dfbe5a324349d95a6bb30ce6d/archinstall/lib/disk/device_handler.py#L245-L258

This list can include existing partitions (seen with `"status": "existing"` in the config) like the one Torxed pointed out above that does not have an `fs_type`. An existing partition being validated is also the cause of the issue #1978.

Filtering the partitions to be validated to only partitions with a status of _modify_ (`ModificationStatus.Modify`) or _create_ (`ModificationStatus.Create`) should fix this unless there is a purpose behind validating partitions with a status of _existing_ (`ModificationStatus.Exist`) or _delete_ (`ModificationStatus.Delete`).

**Comment by poseisharp** [comment]
I have the same issue as OP

**Comment by MXVXID** [comment]
has anyone solved this problem?

**Comment by ManofJELLO** [comment]
Same issue as OP for me

**Comment by poseisharp** [comment]
This is just a workaround that worked for me:
Erased other partition, I only have 2 partition one for /boot and the other for /
and if this is the case manual partitioning worked

However this resulted me nuking my windows installation and made me reinstall it again

So I think the partition does marked as modify but for some reason it still complains about other partition not marked as modify which are not meant to be modified?

**Comment by Faugus** [comment]
@Torxed @svartkanin I found out what's causing this problem for me: the swap partition.

I formatted my nvme and created all the partitions, including the swap partition, using gnome-disk-utility. I tried to install Arch and the same problem occurred.
I formatted again and created only the swap partition, since it can't be created by Archinstall, created the other partitions by Archinstall and tried to install: the same problem occurred.
I formatted again and created all the partitions, except the swap partition, using gnome-disk-utility and tried to install: SUCCESS!
